lib/optimizely/event_builder.rb

```diff
@@ -41,198 +42,202 @@ def ==(event)
+            visitor_id: user_id,
+          }
+        ],
+        revision: @config.revision,
+        client_name: CLIENT_ENGINE,
+        client_version: VERSION,
+      }
+
+      common_params
     end
   end
 
   class EventBuilder < BaseEventBuilder
-    CONVERSION_EVENT_ENDPOINT = 'https://logx.optimizely.com/log/event'
-    IMPRESSION_EVENT_ENDPOINT = 'https://logx.optimizely.com/log/decision'
+    ENDPOINT = 'https://logx.optimizely.com/v1/events'
     POST_HEADERS = { 'Content-Type' => 'application/json' }
+    ACTIVATE_EVENT_KEY = 'campaign_activated'
 
-    def create_impression_event(experiment, variation_id, user_id, attributes)
-      # Create conversion Event to be sent to the logging endpoint.
+    def get_impression_event_params(experiment, variation_id)
+      # Creates object of params specific to impression events
       #
-      # experiment - Experiment for which impression needs to be recorded.
-      # variation_id - ID for variation which would be presented to user.
-      # user_id - ID for user.
-      # attributes - Hash representing user attributes and values which need to be recorded.
+      # experiment -   +Hash+ experiment for which impression needs to be recorded
+      # variation_id - +string+ ID for variation which would be presented to user
       #
-      # Returns event hash encapsulating the impression event.
-
-      @params = {}
-      add_common_params(user_id, attributes)
-      add_decision(experiment, variation_id)
-      add_attributes(attributes)
-      Event.new(:post, IMPRESSION_EVENT_ENDPOINT, @params, POST_HEADERS)
-    end
+      # Returns +Hash+ Impression event params
 
-    def create_conversion_event(event_key, user_id, attributes, event_tags, experiment_variation_map)
-      # Create conversion Event to be sent to the logging endpoint.
-      #
-      # event_key - Event key representing the event which needs to be recorded.
-      # user_id - ID for user.
-      # attributes - Hash representing user attributes and values which need to be recorded.
-      # event_tags - Hash representing metadata associated with the event.
-      # experiment_variation_map - Map of experiment ID to the ID of the variation that the user is bucketed into.
-      #
-      # Returns event hash encapsulating the conversion event.
-
-      @params = {}
-      add_common_params(user_id, attributes)
-      add_conversion_event(event_key)
-      add_event_tags(event_tags)
-      add_layer_states(experiment_variation_map)
-      Event.new(:post, CONVERSION_EVENT_ENDPOINT, @params, POST_HEADERS)
-    end
-
-    private
-
-    def add_common_params(user_id, attributes)
-      super
-      @params['isGlobalHoldback'] = false
-    end
-
-    def add_project_id
-      @params['projectId'] = @config.project_id
-    end
+      experiment_key = experiment['key']
+      experiment_id = experiment['id']
 
-    def add_account_id
-      @params['accountId'] = @config.account_id
-    end
+      decision = [{
```

Comment:
please rename it to decisions as it is array.

Reply:
https://github.com/optimizely/node-sdk/blob/94f940cff1872ec1023fd9eb2ad4a3781d31d61d/lib/core/event_builder/index.js#L109